src/Windows/dll/evtx_logging.cpp

```diff
@@ -0,0 +1,106 @@
+#include "evtx_logging.h"
+
+#define DCAP_EVTX_KEY "SYSTEM\\CurrentControlSet\\Services\\EventLog\\Application\\dcap_quoteprov"
+#define DCAP_DLL "dcap_quoteprov.dll"
+
+DWORD check_install_event_log_source()
```

Comment:
IIRC the windows DCAP code has access to wil (Windows Interface Library) and that contains RAII wrappers for registry keys and registry operations.

It may make sense to use them here to avoid potential leaks.

Reply:
see wil::unique_hkey.